io-engine/src/lvm/vg_pool.rs

```diff
@@ -0,0 +1,294 @@
+                args: args.disks,
+                vg: vg.disks,
+            });
+        }
+
+        if vg.tags.contains(&Property::Lvm) {
+            return Ok(vg);
+        }
+
+        LvmCmd::vg_change(name)
+            .arg("-q")
+            .tag(Property::Lvm)
+            .run()
+            .await?;
+        vg.tags.push(Property::Lvm);
+        Ok(vg)
+    }
+
+    /// Delete the volume group.
+    pub(crate) async fn destroy(mut self) -> Result<(), Error> {
+        // As currently only import of volume group is supported
+        // exporting the volume group on destroy.
+        let name = self.name().to_string();
+        self.export().await?;
+        info!("LVM pool '{name}' has been destroyed successfully");
+        self.ptpl().destroy().ok();
+        Ok(())
+    }
+
+    /// Exports the volume by removing our tag from it.
+    /// todo: export the LV's as well...
+    pub(crate) async fn export(&mut self) -> Result<(), Error> {
+        LvmCmd::vg_change(self.name())
+            .untag(Property::Lvm)
+            .run()
+            .await
+    }
+
+    /// Get the volume group name.
+    pub(crate) fn name(&self) -> &str {
+        self.name.as_str()
+    }
+
+    /// Get the volume group uuid.
+    pub(crate) fn uuid(&self) -> &str {
+        &self.uuid
+    }
+
+    /// Get the volume group disks.
+    pub(crate) fn disks(&self) -> Vec<String> {
+        self.disks.clone()
+    }
+
+    /// Get the volume group capacity.
+    pub(crate) fn capacity(&self) -> u64 {
+        self.size
+    }
+
+    /// Get the volume group committed bytes.
+    pub(crate) fn committed(&self) -> u64 {
+        self.size
+    }
+    /// Get the volume group cluster size.
+    pub(crate) fn cluster_size(&self) -> u64 {
+        4 * 1024 * 1024
+    }
+
+    /// Get the volume group available capacity.
+    pub(crate) fn available(&self) -> u64 {
+        self.free
+    }
+
+    /// Get the volume group used capacity.
+    pub(crate) fn used(&self) -> u64 {
+        self.capacity() - self.available()
+    }
+
+    /// Check if the volume group matches the list options.
+    fn matches(&self, opts: &CmnQueryArgs) -> bool {
+        self.named(opts.name.as_ref())
+            && self.tagged(opts.tag.as_ref())
+            && self.uuided(opts.uuid.as_ref())
+    }
+
+    /// Check if the volume group name matches.
+    fn named(&self, name: Option<&String>) -> bool {
+        let eq = name.map(|name| &self.name == name).unwrap_or(true);
+        tracing::trace!("{name:?} == {} ? {eq}", self.name);
+        eq
+    }
+    /// Check if the volume group contains the given tag.
+    fn tagged(&self, tag: Option<&String>) -> bool {
+        let eq = tag
+            .map(|tag| self.tags.iter().any(|ttag| ttag.key() == tag.as_str()))
+            .unwrap_or(true);
+        tracing::trace!("{tag:?} == {:?} ? {eq}", self.tags);
+        eq
+    }
+    /// Check if the volume group uuid matches.
+    fn uuided(&self, uuid: Option<&String>) -> bool {
```

Comment:
`uuided` goes rhyming with `named` or `tagged` but makes an invalid word :) Perhaps change to `has_uuid` or `uuid_match`

Reply:
is it invalid though? uuid is a noun so we can add "ed" to it?